prov/shm/src/smr_ep.c

```diff
@@ -549,6 +549,10 @@ static int smr_format_sar(struct smr_ep *ep, struct smr_cmd *cmd,
 	pending->bytes_done = 0;
 	pending->next = 0;
 
+	/* Nothing to copy for 0 byte transfer */
+	if (!cmd->msg.hdr.size)
+		return 0;
```

Comment:
Can you update the commit message to indicate *why* SAR needs to handle a 0-byte transfer?  Because on the surface it doesn't make sense.

Reply:
Yeah, it is actually caught by a corner case when I run `fi_rma_bw -p shm -I 5 -v -S 0 -e rdm -s localhost`. In this test, data validation is enabled, and it forces SHM to run with FI_DELIVERY_COMPLETE (DC), and turn off inject. SHM will use SAR always for DC, so if you run fi_send/recv with SHM + DC + 0 byte, this bug can be triggered. I don't know if any real application is doing that though

Reply:
I'd like that information in the commit message, so that's it's captured in the git logs, not the PR text.

Reply:
Done.

Reply:
@shefty do you have more comments?